maintenance/generateTypedDocs.ts

```diff
@@ -0,0 +1,207 @@
+	return file?.getExportedDeclarations().get(identifier)?.[0];
+}
+
+export function stripComments(
+	node: ExportedDeclarations,
+	options: ImportRange["options"],
+): ExportedDeclarations {
+	if (Node.isTextInsertableNode(node)) {
+		// Remove some comments if desired
+		const ranges: { pos: number; end: number }[] = [];
+		const removePredicate = (c: CommentRange) =>
+			(!options.comments &&
+				c.getKind() === SyntaxKind.SingleLineCommentTrivia) ||
+			(!options.jsdoc &&
+				c.getKind() === SyntaxKind.MultiLineCommentTrivia);
+
+		const getCommentRangesForNode = (
+			node: Node,
+		): { pos: number; end: number }[] => {
+			const comments = node.getLeadingCommentRanges();
+			const ret = comments.map((c, i) => ({
+				pos: c.getPos(),
+				end:
+					i < comments.length - 1
+						? comments[i + 1].getPos()
+						: Math.max(node.getStart(), c.getEnd()),
+				remove: removePredicate(c),
+			}));
+			// Only use comment ranges that should be removed
+			return ret.filter((r) => r.remove);
+		};
+
+		if (Node.isEnumDeclaration(node)) {
+			for (const member of node.getMembers()) {
+				ranges.push(...getCommentRangesForNode(member));
+			}
+		} else if (Node.isInterfaceDeclaration(node)) {
+			const walkInterfaceDeclaration = (node: InterfaceDeclaration) => {
+				for (const member of node.getMembers()) {
+					ranges.push(...getCommentRangesForNode(member));
+					if (Node.isInterfaceDeclaration(member)) {
+						walkInterfaceDeclaration(member);
+					}
+				}
+			};
+			walkInterfaceDeclaration(node);
+		}
+
+		// Sort in reverse order, so the removals don't influence each other
+		ranges.sort((a, b) => b.pos - a.pos);
+		for (const { pos, end } of ranges) {
+			node.removeText(pos, end);
+		}
+	}
+	return node;
+}
+
+export function getTransformedSource(
+	node: ExportedDeclarations,
+	options: ImportRange["options"],
+): string {
+	// Remove exports keyword
+	if (Node.isModifierableNode(node)) {
+		node = node.toggleModifier("export", false);
+	}
+
+	// Remove @internal and @deprecated members
+	if (Node.isInterfaceDeclaration(node)) {
+		for (const member of node.getMembers()) {
+			if (
+				member
+					.getJsDocs()
+					.some((doc) =>
+						/@(deprecated|internal)/.test(doc.getInnerText()),
+					)
+			) {
+				member.remove();
+			}
+		}
+	}
+
+	// Comments must be removed last (if that is desired)
+	node = stripComments(node, options);
+
+	// Using getText instead of print avoids reformatting the node
+	let ret = node.getText();
+	// Format with Prettier so we get the original formatting back
+	ret = formatWithPrettier("index.ts", ret).trim();
+	return ret;
+}
+
+interface ImportRange {
+	index: number;
+	end: number;
+	module: string;
+	symbol: string;
+	import: string;
+	options: {
+		comments?: boolean;
+		jsdoc?: boolean;
+	};
+}
+
+const importRegex = /(?<import><!-- #import (?<symbol>.*?) from "(?<module>.*?)"(?: with (?<options>[\w\-, ]*?))? -->)(?:[\s\r\n]*```ts[\r\n]*(?<source>(.|\n)*?)```)?/g;
```

Comment:
@AlCalzone This regex is pure madness :rofl:

Reply:
I've seen worse